pandas/io/json.py

```diff
@@ -884,3 +940,175 @@ def _recursive_extract(data, path, seen_meta, level=0):
         result[k] = np.array(v).repeat(lengths)
 
     return result
+
+
```

Comment:
I think should put here and below (and build_table_schema) in a separate file
``pandas/io/json/schema.py`` (dont forget to add to ``setup.py``).